docs/source/data/fetching-rest.mdx

```diff
@@ -89,75 +126,88 @@ const resolvers = {
 
 ## Caching
 
-By default, data source implementations use Apollo Server's in-memory cache to store the results of past fetches.
+> 📣 **New in Apollo Server 4**: Apollo Server no longer automatically provides it's cache to data sources. [See here for more details](#datasources).
 
-When you initialize Apollo Server, you can provide its constructor a _different_ cache object that implements the [`KeyValueCache` interface](https://github.com/apollographql/apollo-utils/tree/main/packages/keyValueCache#keyvaluecache-interface). This enables you to back your cache with shared stores like Memcached or Redis.
+You can pass your `RESTDataSource` subclasses a cache (such as Apollo Server's default cache) to store the results of past fetches:
 
-### Using an external cache backend
+<MultiCodeBlock>
 
-When running multiple instances of your server, you should use a shared cache backend. This enables one server instance to use the cached result from _another_ instance.
+```ts
+import { RESTDataSource } from '@apollo/datasource-rest';
+// KeyValueCache is the type of Apollo server's default cache
+import { KeyValueCache } from '@apollo/utils.keyvaluecache';
+import { ApolloServer } from '@apollo/server';
+import { startStandaloneServer } from '@apollo/server/standalone';
 
-Apollo Server supports using [Memcached](https://memcached.org/), [Redis](https://redis.io/), or other cache backends via the [`keyv`](https://www.npmjs.com/package/keyv) package. For examples, see [Configuring external caching](../performance/cache-backends#configuring-external-caching).
+class MoviesAPI extends RESTDataSource {
+  baseURL = 'https://movies-api.example.com/';
 
-You can also choose to implement your own cache backend. For more information, see [Implementing your own cache backend](../performance/cache-backends#implementing-your-own-cache-backend).
+  //highlight-start
+  constructor(options: {cache: KeyValueCache }) { 
+    super(options); // this should send our server's `cache` through
+  }
+  //highlight-end
 
-## `RESTDataSource` reference
+  // data fetching methods, etc.
+}
 
-The `RESTDataSource` abstract class helps you fetch data from REST APIs. Your server defines a separate subclass of `RESTDataSource` for each REST API it communicates with.
+interface ContextValue {
+  dataSources: {
+    moviesAPI: MoviesAPI;
+  };
+}
 
-To get started, install the `apollo-datasource-rest` package:
+const server = new ApolloServer<ContextValue>({
+  typeDefs,
+  resolvers,
+});
 
-```bash
-npm install apollo-datasource-rest
+const { url } = await startStandaloneServer(server, {
+  context: async ({ req }) => {
+    const { cache } = server;
+    return {
+      token,
+      //highlight-start
+      dataSources: {
+        moviesAPI: new MoviesAPI({ cache, token }),
```

Comment:
If this is referencing the `MoviesAPI` we created above, then we need to include the token in the constructor signature.

And if we do that, then my previous comment about dropping the default constructor is no longer valid (since `token` is not a part of the base class's constructor signature).